src/DotNetWorker.Grpc/Handlers/InvocationHandler.cs

```diff
@@ -0,0 +1,154 @@
+
+namespace Microsoft.Azure.Functions.Worker.Handlers
+{
+    internal class InvocationHandler : IInvocationHandler
+    {
+        private readonly IFunctionsApplication _application;
+        private readonly IInvocationFeaturesFactory _invocationFeaturesFactory;
+        private readonly IOutputBindingsInfoProvider _outputBindingsInfoProvider;
+        private readonly IInputConversionFeatureProvider _inputConversionFeatureProvider;
+        private readonly ObjectSerializer _serializer;
+        private readonly ILogger _logger;
+
+        private ConcurrentDictionary<string, CancellationTokenSource> _inflightInvocations => new();
+
+        public InvocationHandler(
+            IFunctionsApplication application,
+            IInvocationFeaturesFactory invocationFeaturesFactory,
+            ObjectSerializer serializer,
+            IOutputBindingsInfoProvider outputBindingsInfoProvider,
+            IInputConversionFeatureProvider inputConversionFeatureProvider,
+            ILoggerProvider loggerProvider)
+        {
+            _application = application ?? throw new ArgumentNullException(nameof(application));
+            _invocationFeaturesFactory = invocationFeaturesFactory ?? throw new ArgumentNullException(nameof(invocationFeaturesFactory));
+            _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+            _outputBindingsInfoProvider = outputBindingsInfoProvider ?? throw new ArgumentNullException(nameof(outputBindingsInfoProvider));
+            _inputConversionFeatureProvider = inputConversionFeatureProvider ?? throw new ArgumentNullException(nameof(inputConversionFeatureProvider));
+            _logger = loggerProvider.CreateLogger(nameof(InvocationHandler));
+        }
+
+        public async Task<InvocationResponse> InvokeAsync(InvocationRequest request)
+        {
+            FunctionContext? context = null;
+            InvocationResponse response = new()
+            {
+                InvocationId = request.InvocationId,
+                Result       = new StatusResult()
+            };
+
+            try
+            {
+                var invocation = new GrpcFunctionInvocation(request);
+                var cancellationTokenSource = new CancellationTokenSource();
+
+                if (!_inflightInvocations.TryAdd(request.InvocationId, cancellationTokenSource))
+                {
+                    // Dispose the CancellationTokenSource if we cannot keep track of it
+                    cancellationTokenSource?.Dispose();
+                }
+
+                IInvocationFeatures invocationFeatures = _invocationFeaturesFactory.Create();
+                invocationFeatures.Set<FunctionInvocation>(invocation);
+                invocationFeatures.Set<IExecutionRetryFeature>(invocation);
+
+                var token = cancellationTokenSource?.Token ?? CancellationToken.None;
+                context = _application.CreateContext(invocationFeatures, token);
```

Comment:
I may be missing the reason you have this variable declared above, but I _think_ we can declare it inline here (as well as dispose it):

```suggestion
                using var context = _application.CreateContext(invocationFeatures, token);
```

Reply:
Just copying existing code here, happy to create an issue for cleaning up some of the old stuff after I finish the cancellation work